File: domain/src/main/java/org/openmedstack/eventstore/ISaveSnapshots.java
package org.openmedstack.eventstore;

import org.openmedstack.domain.AggregateRootBase;
import org.openmedstack.domain.Memento;

import java.util.concurrent.CompletableFuture;

public interface ISaveSnapshots {
    <T extends Memento> CompletableFuture<Boolean> save(AggregateRootBase<T> aggregateRoot);
}

File: modules/domain/src/main/kotlin/org/openmedstack/eventstore/ISaveSnapshots.kt
package org.openmedstack.eventstore

import org.openmedstack.domain.AggregateRootBase
import org.openmedstack.domain.Memento
import java.util.concurrent.CompletableFuture

interface ISaveSnapshots {
    fun <T : Memento> save(aggregateRoot: AggregateRootBase<T>): CompletableFuture<Boolean>
}
